File: KnowRad/src/main/java/br/com/knowrad/service/study/ModalidadeService.java
package br.com.knowrad.service.study;

import br.com.knowrad.dto.study.ModalidadeDTO;
import br.com.knowrad.entity.study.Modalidade;

import java.util.List;

public interface ModalidadeService {

    List<Modalidade> findAll();

    List<ModalidadeDTO> findAllDTO();

    Modalidade findById(Long id);

    List<ModalidadeDTO> search(String searchText, Integer limit);
}

File: KnowRad/src/main/webapp/assets/js/knowrad/study/modalidade-service.js
"use strict"

var ModalidadeService = {

    service: new AbstractService("modalidade/"),

    /**
     * Traz todas as modalidades do banco
     * @param callback
     */
    findAllDTO: function(callback) {

        this.service.doGET("findAllDTO", ({}), callback);

    },

    search: function(searchText, limit, callback) {

        this.service.doGET("search", ({searchText: searchText, limit: limit}), callback);

    }

};